src/box/memtx_tx.c

```diff
@@ -1765,6 +1775,24 @@ memtx_tx_history_prepare_insert_stmt(struct txn_stmt *stmt)
 			memtx_tx_handle_conflict(stmt->txn, tracker->reader);
 		}
 	}
+
+	for (uint32_t i = 0; i < index_count; i++) {
```

Comment:
Could you please supply this part with comment or mb even move to a helper function?

Reply:
I'll explain here. If it is useful I think we could make a follow up.
The thing before this  patch read trackers could be stored only in prepared stories. After this patch they can appear in any place in story chain. That's why we have to walk the chain again and collect those conflicts during prepare.